Gem/Nodes/ViewModels/ImportNodeSceneViewModel.swift

```diff
@@ -0,0 +1,139 @@
+// Copyright (c). Gem Wallet. All rights reserved.
+
+import SwiftUI
+import Primitives
+import Components
+import BigInt
+
+class ImportNodeSceneViewModel: ObservableObject {
+    private let nodeService: NodeService
+    private let importNodeService: ImportNodeService
+
+    @Published private var chain: Chain
+
+    @Published var urlString: String = ""
+    @Published var state: StateViewType<Bool> = .noData
+    @Published var chainID: String = ""
+    @Published var blockNumber: String?
+    @Published var isInSync: Bool = false
+
+    init(chain: Chain, nodeService: NodeService) {
+        self.chain = chain
+        self.nodeService = nodeService
+        self.importNodeService = ImportNodeService(nodeStore: nodeService.nodeStore)
+    }
+
+    var shouldDisableImportButton: Bool {
+        return state.isLoading || state.isNoData || state.isError || !isInSync
+    }
+
+    var title: String {
+        Localized.Settings.Networks.ImportNode.title
+    }
+}
+
+// MARK: - Business Logic
+
+extension ImportNodeSceneViewModel {
+    func importFoundNode()  {
+        let node = Node(url: urlString, status: .active, priority: 10)
+        try? importNodeService.importNode(ChainNodes(chain: chain.rawValue, nodes: [node]))
+        try? nodeService.setNodeSelected(chain: chain, node: node)
+    }
+
+    func getNetworkInfo() async throws  {
+        guard let url = URL(string: urlString) else {
+            await updateStateWithError()
+            return
+        }
+
+        await MainActor.run { [self] in
+            self.state = .loading
+        }
+
+        let provider = ChainServiceFactory(nodeProvider: CustomNodeULRFetchable(url: url))
+        let service = provider.service(for: chain)
+
+        async let isNodeSync = service.getInSync()
+        async let chainId = service.getChainID()
+        async let blockNumber = service.getLatestBlock()
+
+        do {
+            let (isSynced, networkId, blockNumber) = try await (isNodeSync, chainId, blockNumber)
+
+            guard networkId == getPrimaryNetworkId(for: chain) else {
+                await updateStateWithError()
+                return
+            }
+
+            await MainActor.run { [self] in
+                self.isInSync = isSynced
+                self.chainID = networkId
+                self.blockNumber = blockNumber
+                self.state = .loaded(true)
+            }
+        } catch {
+            await updateStateWithError()
+        }
+    }
+}
+
+// MARK: - Private
+
+extension ImportNodeSceneViewModel {
+    private func updateStateWithError() async {
+        await MainActor.run {
+            self.state = .error(AnyError(Localized.Errors.invalidUrl))
+        }
+    }
+
+    private func getPrimaryNetworkId(for chain: Chain) -> String {
```

Comment:
use `StakeConfig.config()` it will return config for each chain that will contain `network_id`

Reply:
if `network_id` optional, do not check network id field